lib/swarm.js

```diff
@@ -3,46 +3,36 @@ var subleveldown = require('subleveldown')
 var swarm = require('webrtc-swarm')
 var signalhub = require('signalhub')
 var hyperlog = require('hyperlog')
-var ghsign = require('ghsign')
 var through = require('through2')
 var protobuf = require('protocol-buffers')
 var fs = require('fs')
-var request = require('request')
 
 var messages = protobuf(fs.readFileSync(__dirname + '/../schema.proto'))
 
 module.exports = createSwarm
 
-ghsign = ghsign(function (username, cb) {
-  fs.readFile('./public-keys/' + username + '.keys', 'utf-8', function (_, keys) {
-    if (keys) return cb(null, keys)
-    request('https://github.com/' + username + '.keys', function (_, response) {
-      var keys = response.statusCode === 200 && response.body
-      if (!keys) return cb(new Error('Could not find public keys for ' + username))
-      fs.mkdir('./public-keys', function () {
-        fs.writeFile('./public-keys/' + username + '.keys', keys, function () {
-          cb(null, keys)
-        })
-      })
-    })
-  })
-})
-
-var verifiers = {}
-
 function createSwarm (db, defaultOpts) {
   var emitter = new events.EventEmitter()
   var logs = {}
 
   emitter.logs = logs
 
-  var sign
   var processor
+  var sign
+  var verify
 
   emitter.changes = function (name) {
     return logs[name] ? logs[name].changes : 0
   }
 
+  emitter.sign = function (fn) {
+    sign = fn
+  }
+
+  emitter.verify = function (fn) {
+    verify = fn
+  }
+
```

Comment:
These are to pass in functions to `sign` and `verify` messages.

`swarm.js` is running these because the signed/verified data is actualy the `protobuf`-encoded message itself, which is not readily accessible outside the module.


More info in my comment: https://github.com/moose-team/friends/pull/129#issuecomment-103314306

I do agree that the variable name `emitter` is a little confusing. If it were named `swarm` it would be a bit more intuitive I think.